.github/workflows/flake8.yml

```diff
@@ -0,0 +1,29 @@
+name: Flake8
+
+on:
+  push:
+    branches: [ main ]
+  pull_request:
+    branches: [ main ]
+
+jobs:
+  build:
+
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@v3
+
+    - name: Set up Python 3.8
+      uses: actions/setup-python@v4
+      with:
+        python-version: '3.8'
+
+    - name: Install pre-commit hooks
```

Comment:
Isn't the purpose of pre-commit is to check for error before committing the change in the first place ? Do we need to also set it and execute it also in the pipeline?

Reply:
Yes, because you cannot enforce it locally. A user can always submit without activating `pre-commit` (e.g. it happened also accidentally to me already).

Reply:
I see makes sense thanks.